src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/X509Pal.PublicKey.cs

```diff
@@ -51,6 +56,50 @@ public AsymmetricAlgorithm DecodePublicKey(Oid oid, byte[] encodedKeyValue, byte
             }
         }
 
+        private static ECDsa DecodeECDsaPublicKey(byte[] encodedKeyValue)
+        {
+            if (encodedKeyValue.Length < 1 || (AsnTag)(encodedKeyValue[0]) != AsnTag.OCTET_STRING)
+                throw new CryptographicException(SR.InvalidPublicKeyInX509);
+
+            // After the OCTET_STRING tag, the encryptedKeyValue are X, followed by Y, in big-endian format.
```

Comment:
Sort of... not really... it depends.
```
   SubjectPublicKeyInfo  ::=  SEQUENCE  {
        algorithm            AlgorithmIdentifier,
        subjectPublicKey     BIT STRING  }

   AlgorithmIdentifier  ::=  SEQUENCE  {
        algorithm               OBJECT IDENTIFIER,
        parameters              ANY DEFINED BY algorithm OPTIONAL  }
```

EC keys are stored very complicatedly/polymorphicly.  For Windows 8.1 we only support the parameter being the OID for P256, P384, or P521, and then the blob should be 0x04 (uncompressed key, unless it isn't...) then Qx then Qy.

But, really, why are you reading this at all?  Just import the whole public key subject info block into BCrypt and return a ECDsaCng.

We don't really want to have to write decoders for named uncompressed keys, named compressed keys, specified uncompressed keys, and specified compressed keys.

But this code is just assuming it's a named (and supported) uncompressed key, which is bad and dangerous.